accord-core/src/main/java/accord/coordinate/Invalidate.java

```diff
@@ -211,10 +218,10 @@ else if (homeKey != null)
     @Override
     public void onFailure(Id from, Throwable failure)
```

Comment:
This is something defensive to ensure it's considered by the implementor - which generally it wasn't being. I'm not sure what your concern is about `isDone` - the idea here is we don't want to invoke the callback multiple times...

Reply:
OK, I think I get what you mean. We don't want to give the callback a chance to process the error if the error came from the callback (the one passed to `Invalidate`).

The way it works now it just looks like there are a few cases where an error occurs for the first time and we don't do anything with it in `onCallbackFailure` because `isDone` what set to true somewhere in `onFailure` or `oSuccess`.